frugalos_segment/src/client/storage.rs

```diff
@@ -760,23 +780,36 @@ impl Future for GetReplicatedFragment {
     }
 }
 
+/// Reconstructs original data from dispersed fragments even if
+/// a focusing node loses his data fragment.
 pub struct GetDispersedFragment {
+    /// The processing order of futures
     phase: Phase<CollectFragments, BoxFuture<Vec<u8>>>,
+
+    /// A thread pool of encoders(by erasure code)
     ec: ErasureCoderPool<LibErasureCoderBuilder>,
-    missing_index: usize,
+
+    /// The index of a focusing node.
+    /// None represents that there is no missing index.
+    missing_index: Option<usize>,
 }
 impl Future for GetDispersedFragment {
-    type Item = Vec<u8>;
+    type Item = MaybeFragment;
     type Error = Error;
     fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
+        let missing_index = match self.missing_index {
+            Some(index) => index,
+            None => return Ok(Async::Ready(MaybeFragment::NotParticipant)),
+        };
+
```

Comment:
I suggest
```rust
if self.missing_index.is_none() {
  return Ok(Async::Ready(MaybeFragment::NotParticipant));
}
let missing_index = self.missing_index.unwrap();
...
```
(But I don't know whether Clippy allows the above code ;-( )

Reply:
fixed. clippy allows `unwrap`(and `expect`).

Reply:
IMO: I like the previous non unwrap/expect version (it is not needed to restore the previous code).